### ISSUE TITLE
Installer doesn't remove old dhcpcd configs

### ISSUE THREAD

**Issue by coaxial** [pi-hole/pi-hole#1976]
**In raising this issue, I confirm the following:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

1

---
**Expected behaviour:**

The subnet mask to be respected, and the default gateway to be as entered during the wizard.

**Actual behaviour:**

Subnet mask is `/24`, default gateway is 192.168.1.1

**Steps to reproduce:**

```bash
# set pihole up with static IP 192.168.1.2/24 and gateway 192.168.1.1
# then update the config with the reconfigure option:
$ pihole -r
# go through the wizard, set the static IP to be 192.168.0.2/30 and the gateway 192.168.0.1
# reboot
$ ip a
# shows the IP as 192.168.0.2/24
$ ip r
# shows the default gateway as 192.168.1.1
```

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

`no internet connectivity, can't upload :(`

**Troubleshooting undertaken, and/or other relevant information:**

`{Steps of what you have done to fix this}`

**Comment by jacobsalmela** [comment]
I'm a bit confused with your question.  Why are you setting the IP to 192.168.0.2/30 when your Pi-hole when you have already defined it as 192.168.1.2?

**Comment by dschaper** [comment]
What is the contents of the bottom of `/etc/dhcpcd.conf` where the static IP address is set?

**Comment by coaxial** [comment]
@jacobsalmela My network configuration has changed, the Pi-hole now sits on a different subnet.

@dschaper It looks like it got amended improperly by the script:
```conf
# A hook script is provided to lookup the hostname if not set by the DHCP
# server, but it should not be run by default.
nohook lookup-hostname
## interface eth0
        static ip_address=192.168.1.2/24
        static routers=192.168.1.1
        static domain_name_servers=192.168.1.1
interface eth0
  static ip_address=192.168.0.2/24
  static routers=192.168.0.1
  static domain_name_servers=127.0.0.1
interface eth0
  static ip_address=192.168.0.2/30
  static routers=192.168.0.1
  static domain_name_servers=127.0.0.1
```

**Comment by dschaper** [comment]
There should only be one `static ip_address` entry, remove the other two and reboot the device.

**Comment by coaxial** [comment]
Thank you. It's still a bug though, I didn't manually edit the file; the `pihole -r` command did.

**Comment by technicalpyro** [comment]
> Thank you. It's still a bug though, I didn't manually edit the file; the pihole -r command did.

just to confirm you selected repair or reconfigure?

**Comment by coaxial** [comment]
I'm sorry, I can't remember; it's been a while and I didn't have to go through these steps again recently.

**Comment by mfreema5** [comment]
> just to confirm you selected repair or reconfigure?

I just stumbled my way through this problem (multiple entries in `/etc/dhcpcd.conf`), and I am confident I used the “reconfigure” option when I ran `pinhole -r`.

**Comment by billthefarmer** [comment]
I can confirm multiple entries in `/etc/dhcpcd.conf` on `pihole -r` 'reconfigure'. Also `pihole uninstall` does not remove entries.

**Comment by dweremeichik** [comment]
I ran into this one as well - I set up the pi-hole on my network, with the intention of bringing it to a family member's network. The IP range and gateway were of course different on that network. It's an easy fix if you know what your looking for. Not so if you haven't worked with Linux much.

**Comment by dschaper** [comment]
It's not quite as easy a fix. During installation we ask specifically what IP address to set so we know what to add to `dhcpcd.conf`. Any edits or changes to that file after install finishes can destructively remove intended changes. We don't know if a user made changes to that file or another process has modified that file. Personally I'd like to just not be involved in the IP addressing and tell users to set that up independently before installation.

**Comment by dweremeichik** [comment]
I hear you - and that makes sense. My comment on the easy fix part was that of the user and not the maintainers. I understand that its a pain in the neck and can be brittle - didn't mean to come off the wrong way. 

I took a look into the code and see what is going on. I noticed that there is a "static IP check" but it only checks to make sure that the IP you are currently trying to set isn't already there.

Perhaps a grep like the following would be better:
```regex
^\s*static ip_address
```
Then if the test matches - tell the user what file they need to edit manually to point them in the correct direction.

Something similar would need to be done for ifcfg as well.

For anyone else if they are curious: https://github.com/pi-hole/pi-hole/blob/e41c4b5bb691cea1f5b950d39518d8c404b5846e/automated%20install/basic-install.sh#L821-L880
